File: YRUserComponentImpl/src/main/java/com/apemans/usercomponent/baseinfo/utils/CollectionUtil.java
package com.apemans.usercomponent.baseinfo.utils;

import java.util.Collection;
import java.util.Collections;
import java.util.List;

public class CollectionUtil {
    private CollectionUtil() {
    }

    public static boolean isEmpty(Collection collection) {
        return null == collection || collection.isEmpty();
    }

    public static boolean isNotEmpty(Collection collection) {
        return null != collection && !collection.isEmpty();
    }

    public static boolean isIndexSafe(int index, int size) {
        return index >= 0 && index < size;
    }

    public static <T> List<T> safeFor(List<T> other) {
        return other == null ? Collections.EMPTY_LIST : other;
    }

    public static List emptyList() {
        return Collections.emptyList();
    }

    public static <T> int size(List<T> other) {
        return other == null ? 0 : other.size();
    }
}

File: YRBase/src/main/java/com/apemans/base/utils/CollectionUtil.kt
/*
 * * * Copyright (c) 2021 海龙 Inc. All rights reserved.
 */

package com.apemans.base.utils

class CollectionUtil {
    companion object {
        @JvmStatic
        fun isEmpty(collection: Collection<*>?): Boolean {
            return null == collection || collection.isEmpty()
        }

        @JvmStatic
        fun isNotEmpty(collection: Collection<*>?): Boolean {
            return null != collection && !collection.isEmpty()
        }

        @JvmStatic
        fun isIndexSafe(index: Int, size: Int): Boolean {
            return index >= 0 && index < size
        }

        @JvmStatic
        fun emptyList(): List<*> {
            return emptyList<Any>()
        }

        @JvmStatic
        fun <T> size(other: List<T>?): Int {
            return other?.size ?: 0
        }

        @JvmStatic
        fun <T> safeFor(other: List<T>?): List<T> {
            if (other == null) {
                return mutableListOf()
            } else {
                return other
            }
        }
    }

}

